### ISSUE TITLE
Unexpected token at } catch { error in BitBucket pipeline

### ISSUE THREAD

**Issue by chilupa** [eslint/eslint#14043]
**Tell us about your environment**

* **ESLint Version:**  v7.18.0
* **Node Version:** v12.20.1
* **npm Version:** v6.14.10

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    es2021: true,
  },
  extends: ['plugin:react/recommended', 'airbnb'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 12,
    sourceType: 'module',
  },
  plugins: ['react'],
  rules: {
    'react/jsx-filename-extension': [1, { extensions: ['.js', '.jsx'] }],
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

In my React application, `lint` is one of the scripts in `package.json`.
```js
  "scripts": {
    "lint": "./node_modules/.bin/eslint src"
  },
```
I built a BitBucket pipeline to perform lint checks using the below command 

```bash
eslint src
```

**What did you expect to happen?**
I expected the command `eslint src` would do lint checks on all the files in the `src` folder.

**What actually happened? Please include the actual, raw output from ESLint.**
I got this error in the BitBucket's pipeline.
```bash
> eslint src
/opt/atlassian/pipelines/agent/build/node_modules/eslint/bin/eslint.js:93
        } catch {
                ^
SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:607:28)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Function.Module.runMain (module.js:684:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! funretro@0.1.0 lint: `eslint src`
npm ERR! Exit status 1
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by anikethsaha** [comment]
Thanks for the report.

Whats the nodejs version running in your bitbucket ?
this is an [optional catch binding feature](https://github.com/tc39/proposal-optional-catch-binding). I guess this is supported  from 12 without any flags ! not sure.

**Comment by chilupa** [comment]
@anikethsaha Thank you for the response. I specified the node version to `12` in the `image` and that fixed the issue. 🎉 

Here's what I did for my **BitBucket** pipeline. 
```
image: node:12

pipelines:
  default:
    - parallel:
        - step:
            name: 'Build and Test'
            script:
              - echo "Your build and test goes here..."
        - step:
            name: 'Lint'
            script:
              - npm install
              - npm run lint
```

**Comment by aladdin-add** [comment]
Closing this issue as it looks like the question has been answered. If you still need help, please send
a message to our [mailing list](https://groups.google.com/group/eslint) or
[chatroom](https://eslint.org/chat). Thanks!
